Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/transformation/TransformationManager.java

```diff
@@ -23,17 +26,25 @@
     @Autowired
     private ITransformFilesDAO transformationDAO;
 
+    @Autowired
+    private ITransformationSaveService transformationFileService;
+
     @Transactional
     @Override
-    public void saveTransformation(String title, String description,
-            String patternFileName, String patternTitle,
-            String patternDescription, String mappingFileName,
-            String mappingTitle, String mappingDescription,String userName) {
-        TransformFilesDTO tranformDTO = new TransformFilesDTO(title,
-                description, patternFileName, patternTitle, patternDescription,
-                mappingFileName, mappingTitle, mappingDescription, userName);
+    public void saveTransformations(ITransformationFile transformationFile) throws FileStorageException {
+        TransformFilesDTO tranformDTO = new TransformFilesDTO(transformationFile.getTitle(),
+                transformationFile.getDescription(), transformationFile.getPatternFileName(),
+                transformationFile.getPatternTitle(), transformationFile.getPatternDescription(),
+                transformationFile.getMappingFileName(), transformationFile.getMappingTitle(),
+                transformationFile.getMappingDescription(), transformationFile.getUserName());
         tranformDTO.setId(transformationDAO.generateUniqueID());
         transformationDAO.saveNewDTO(tranformDTO);
+
+        try {
+            transformationFileService.saveFileToLocal(transformationFile);
+        } catch (FileStorageException fs) {
+            throw new FileStorageException(fs);
```

Comment:
there is no point in doing this. what would be the reason to catch an exception just to throw the same type?